src/serac/physics/solid_functional.hpp

```diff
@@ -324,24 +335,49 @@ class SolidFunctional<order, dim, Parameters<parameter_space...>, std::integer_s
   {
     residual_->AddDomainIntegral(
         Dimension<dim>{},
-        DependsOn<0, 1, active_parameters + 2 ...>{},  // the magic number "+2" accounts for the fact that the
-                                                       // displacement and acceleration fields are always-on and come
-                                                       // first, so the `n`th parameter will actually be argument `n+2`
-        [this, material](auto /*x*/, auto& state, auto displacement, auto acceleration, auto... params) {
+        DependsOn<0, 1, 2,
+                  active_parameters + 3 ...>{},  // the magic number "+2" accounts for the fact that the
+                                                 // displacement and acceleration fields are always-on and come
+                                                 // first, so the `n`th parameter will actually be argument `n+2`
+        [this, material](auto /*x*/, auto& state, auto displacement, auto acceleration, auto shape, auto... params) {
           auto du_dX   = get<DERIVATIVE>(displacement);
           auto d2u_dt2 = get<VALUE>(acceleration);
-          auto stress  = material(state, du_dX, params...);
+          auto dp_dX   = get<DERIVATIVE>(shape);
+
+          // Compute the displacement gradient with respect to the shape-adjusted coordinate.
+
+          // Note that the current configuration x' = X + u + p, where X is the original reference
```

Comment:
We talked about this in person, so this comment is for record keeping: this is getting pretty involved with all of the change-of-configuration mappings. I support the suggestion to move to Cauchy or Piola as our expectation from the constitutive.

Reply:
I put this down as an issue. I agree we should move to either Cauchy or Piola in a follow on PR.